### PR TITLE
Introducing semver + "check for updates"

### PR DESCRIPTION
**Pull request by mieciu**
This PR adds simple "check for updates" functionality. 

We take the latest release from GitHub release API and compare it to currently running version. If new version is available, we log a warning and display beautiful yellow banner in the UI:

image

(keeping my FE skills sharp 💪)

### CODE REVIEW

File: quesma/buildinfo/build.go [hunk 1]
```diff
@@ -2,6 +2,73 @@
...
+	}
+	return vCurrent.LessThan(*vLatest), nil
+}
+
+// CheckForTheLatestVersion obtains the latest release information from GitHub and compares it to the currently running version.
+// It returns a user-facing message indicating whether the latest version is newer, the same, or if there was an error.
+func CheckForTheLatestVersion() (updateAvailable bool, messageBanner string) {
+	latestRelease, err := getLatestRelease()
+	if err != nil {
+		return updateAvailable, fmt.Sprintf("Failed obtaining latest Quesma version form GitHub: %v", err)
```
Comment (avelanarius) [inline quesma/buildinfo/build.go:50]:
```suggestion
		return updateAvailable, fmt.Sprintf("Failed obtaining the latest Quesma version from GitHub: %v", err)
```

File: quesma/main.go [hunk 2]
```diff
@@ -59,6 +59,9 @@ func main() {
 	}, sig, doneCh, asyncQueryTraceLogger)
 	defer logger.StdLogFile.Close()
 	defer logger.ErrLogFile.Close()
+	if upgradeAvailable, message := buildinfo.CheckForTheLatestVersion(); upgradeAvailable {
```
Comment (mieciu) [inline quesma/main.go:62]: I might actually put that into separate gorountine

File: quesma/buildinfo/build.go [hunk 3]
```diff
@@ -2,6 +2,73 @@
...
+	}
+	return vCurrent.LessThan(*vLatest), nil
+}
+
+// CheckForTheLatestVersion obtains the latest release information from GitHub and compares it to the currently running version.
+// It returns a user-facing message indicating whether the latest version is newer, the same, or if there was an error.
+func CheckForTheLatestVersion() (updateAvailable bool, messageBanner string) {
+	latestRelease, err := getLatestRelease()
+	if err != nil {
+		return updateAvailable, fmt.Sprintf("Failed obtaining latest Quesma version from GitHub: %v", err)
```
Comment (avelanarius) [inline quesma/buildinfo/build.go:50]:
```suggestion
		return false, fmt.Sprintf("Failed obtaining latest Quesma version from GitHub: %v", err)
```
and similarly below?

Comment (mieciu) [reply quesma/buildinfo/build.go]: https://github.com/QuesmaOrg/quesma/pull/543/commits/c1a5112ddef5f8e28f777fe296869152d5871d41

**Review by avelanarius — APPROVED** [review]

**Comment by mieciu** [comment]
Added small tweak in https://github.com/QuesmaOrg/quesma/pull/543/commits/9f08e31dda45bcc580d6615d0be47620479513c9 - we're still blocking on a call to GitHub, yet I have introduced a timeout so that the whole UI won't hang on GitHub outage

**Pull request merged**